src/content/reference/react/useOptimistic.md

```diff
@@ -0,0 +1,202 @@
+---
+title: useOptimistic
+canary: true
+---
+
+<Canary>
+
+The `useOptimistic` Hook is currently only available in React's canary and experimental channels. Learn more about [React's release channels here](/community/versioning-policy#all-release-channels).
+
+</Canary>
+
+<Intro>
+
+`useOptimistic` is a React Hook that lets you optimistically update the UI.
+
+```js
+  const [optimisticState, addOptimistic] = useOptimistic(state, updateFn);
+```
+
+</Intro>
+
+<InlineToc />
+
+---
+
+## Reference {/*reference*/}
+
+### `useOptimistic(state, updateFn)` {/*use*/}
+
+`useOptimistic` is a React Hook that lets you optimistically update the UI.
+
+```js
+import { useOptimistic } from 'react';
+
+function AppContainer() {
+  const [optimisticState, addOptimistic] = useOptimistic(
+    state,
+    // updateFn
+    (currentState, newState) => {
+      // merge and return new state
+    }
+  );
+}
+```
+
+[See more examples below.](#usage)
+
+#### Parameters {/*parameters*/}
+
+* `state`: `state` is the source of truth so if the `state` is ever changed and we get a new value in from the server, the return value `optimisticState` will be set the that too since it will always treat `state` as the final source of truth.
+* `updateFn`: `updateFn` is the mutation that will occur to `optimisticState` when `addOptimistic` is called. `updateFn` takes in two parameters. The `currentState` and the `newState`. The return value will be the merged value of the `currentState` and `newState`.
+
+
+#### Returns {/*returns*/}
+
+* `optimisticState`: `optimisticState` is the optimistic state, it will default to what `state` is.
+* `addOptimistic`: `addOptimistic` is the dispatching function to call that will run what we define in `updateFn`.
```

Comment:
```suggestion
* `addOptimistic`: `addOptimistic` is the dispatching function to call when you have an optimistic update. It takes one argument, `optimisticValue`, of any type and will call the `updateFn` with `state` and `optimisticValue`.
```

Alternatively we could flush this section out like we do in `setState` like @mattcarrollcode mentioned: https://react-dev-git-fork-kmiddleton14-use-optimistic-fbopensource.vercel.app/reference/react/useState#setstate -- like how it creates another type signature explanation for the setter

Reply:
That's a good idea and one I can add in the second iteration